src/xcode/ENA/ENA/Source/Scenes/Checkin/CheckInDetail/TraceLocationDetailViewModel.swift

```diff
@@ -0,0 +1,122 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import UIKit
+import OpenCombine
+
+final class TraceLocationDetailViewModel {
+
+	// MARK: - Init
+	
+	init(_ traceLocation: TraceLocation) {
+		self.traceLocation = traceLocation
+	}
```

Comment:
```suggestion
	init(_ traceLocation: TraceLocation) {
		self.traceLocation = traceLocation
		self.locationDescription = traceLocation.description
 		self.locationAddress = traceLocation.address
		
		selectedDurationInMinutes = initialDuration
	}
```

If you move these here, you can keep them in the viewModel and keep the traceLocation private. And you can get rid of `setupViewModel()`. And instead of the `locationInitialDuration` you can just directly use selectedDurationInMinutes.

Reply:
so we keep the locationDescription as an internal property in the viewModel ?because we still need to access it from the VC

Reply:
it is done now